mdpow/equil.py

```diff
@@ -361,13 +396,26 @@ def solvate(self, struct=None, **kwargs):
         kwargs['bt'] = boxtype
 
         kwargs['includes'] = asiterable(kwargs.pop('includes',[])) + self.dirs.includes
-
-        params = gromacs.setup.solvate(**kwargs)
```

Comment:
I would use inheritance instead of `if self.solvent_type == 'wetoctanol':`: Replace the line 365 with
```python
params = self._setup_solvate(**kwargs)
```
and then put in the base class
```python
def _setup_solvate(**kwargs):
   return gromacs.setup.solvate(**kwargs)
```
but in the WetOctanolSimulation class override

```python
class WetOctanolSimulation(Simulation):
   def  _setup_solvate(**kwargs):
            sol = gromacs.setup.solvate_sol(**kwargs)
            with in_dir(self.dirs.solvation, create=False):
                u = mda.Universe('solvated.gro')
                octanol = u.select_atoms('resname OcOH')
                n = octanol.n_residues
            with in_dir(self.dirs.topology, create=False):
                gromacs.cbook.edit_txt(self.top_template,
                                       [('OcOH               1', '1', n)])
            ionkwargs = kwargs
            ionkwargs['struct'] = sol['struct']
            params = gromacs.setup.solvate_ion(**ionkwargs)
            return params
```

This keeps all the specialization in its own class and makes it easier to extend the code.